src/api/entities/HistoricalMultiSigProposal.ts

```diff
@@ -0,0 +1,177 @@
+  rejectionCount: number;
+  creator?: {
+    did: string;
+  };
+  status: string;
+  createdBlockId: string;
+  updatedBlockId: string;
+  datetime: Date;
+};
+
+type MiddlewareMultiSigProposalUniqueIdentifiers = Pick<MiddlewareMultiSigProposal, 'id'>;
+
+export interface HumanReadable {
+  id: string;
+  proposalId: number;
+  multisigId: string;
+  approvalCount: number;
+  rejectionCount: number;
+  creatorDid?: string;
+  status: string;
+  createdBlockId: string;
+  updatedBlockId: string;
+  datetime: string;
+}
+
+enum MultiSigStatusEnum {
+  Pending = 'Pending',
+  Success = 'Success',
+  Failed = 'Failed',
+}
+
+/**
+ * Represents historical MultiSigProposal that no longer exists on chain
+ */
+export class HistoricalMultiSigProposal extends Entity<
+  MiddlewareMultiSigProposalUniqueIdentifiers,
+  HumanReadable
+> {
+  public id: string;
+  public proposalId: BigNumber;
+  public multisigId: string;
+  public approvalCount: BigNumber;
+  public rejectionCount: BigNumber;
+  public creatorDid?: string;
+  public status: MultiSigStatusEnum;
+  public createdBlockId: string;
+  public updatedBlockId: string;
+  public datetime: Date;
+
+  /**
+   * @hidden
+   */
+  public constructor(
+    identifiers: MiddlewareMultiSigProposalUniqueIdentifiers,
+    proposal: MiddlewareMultiSigProposal,
+    context: Context
+  ) {
+    const { id } = identifiers;
+
+    const {
+      proposalId,
+      multisigId,
+      approvalCount,
+      rejectionCount,
+      creator,
+      status,
+      createdBlockId,
+      updatedBlockId,
+      datetime,
+    } = proposal;
+    super({ id }, context);
+
+    this.id = id;
+    this.proposalId = new BigNumber(proposalId);
+    this.multisigId = multisigId;
+    this.approvalCount = new BigNumber(approvalCount);
+    this.rejectionCount = new BigNumber(rejectionCount);
+    this.creatorDid = creator?.did;
+    this.status = status as MultiSigStatusEnum;
+    this.createdBlockId = createdBlockId;
+    this.updatedBlockId = updatedBlockId;
+    this.datetime = datetime;
+  }
+
+  /**
+   * @hidden
+   * Check if a value is of type {@link MiddlewareMultiSigProposalUniqueIdentifiers}
+   */
+  public static override isUniqueIdentifiers(
+    identifier: unknown
+  ): identifier is MiddlewareMultiSigProposalUniqueIdentifiers {
+    const { id } = identifier as MiddlewareMultiSigProposalUniqueIdentifiers;
+
+    return typeof id === 'string';
+  }
+
+  /**
+   * @hidden
+   */
+  public exists(): Promise<boolean> {
+    return Promise.resolve(false);
+  }
+
+  /**
+   * Determine whether this Entity is the same as another one
+   */
+  public override isEqual(entity: HistoricalMultiSigProposal): boolean {
+    return entity.id === this.id;
+  }
+
+  /**
+   * Get the MultiSigProposalVotes associated with this MultiSigProposal
+   */
+  public async getVotes(): Promise<MultiSigProposalVote[]> {
+    const {
+      context: { queryMiddleware },
+      id: proposalId,
+      context,
+    } = this;
+
```

Comment:
This can be added as a method named `getVotes` in MultiSigProposal entity

Reply:
I double checked and this is already present as method named `votes` in `MultiSigProposal` entity